index.js

```diff
@@ -641,16 +641,20 @@ function forAwaitEach(source, callback, thisArg) {
   var asyncIterator = createAsyncIterator(source)
   if (asyncIterator) {
     var i = 0
-    function next() {
-      return asyncIterator.next().then(function(step) {
-        if (!step.done) {
-          return Promise.resolve(
-            callback.call(thisArg, step.value, i++, source)
-          ).then(next)
-        }
-      })
-    }
-    return next()
+    return new Promise(resolve => {
```

Comment:
You are not handling rejecting as well,
this cause Unhandled promise rejection if asycIterator fails..

Reply:
Good catch @DxCx do you think https://github.com/leebyron/iterall/pull/28 will address that?